src/models/BanList.ts

```diff
@@ -96,6 +143,36 @@ export default class BanList {
                 continue; // invalid/unknown
             }
 
+            const previousState = this.getState(event['type'], event['state_key']);
+            this.setState(event['type'], event['state_key'], event);
+            const changeType: null|ChangeType = (() => {
+                if (!previousState) {
+                    return ChangeType.Added;
+                } else if (previousState['event_id'] === event['event_id']) {
+                    if (event['unsigned']?.['redacted_because']) {
+                        return ChangeType.Removed;
+                    } else {
+                        // Nothing has changed.
+                        return null;
+                    }
+                } else {
+                    // Then the policy has been modified in some other way, possibly 'soft' redacted by a new event with empty content...
+                    if (Object.keys(event['content']).length === 0) {
+                        return ChangeType.Removed;
+                    } else {
+                        return ChangeType.Modified;
+                    }
+                }
+            })();
+
+            // If we haven't got any information about what the rule used to be, then it wasn't a valid rule to begin with
+            // and so will not have been used. Removing a rule like this therefor results in no change.
+            if (changeType === ChangeType.Removed && previousState?.unsigned?.rule) {
+                const sender = event.unsigned['redacted_because'] ? event.unsigned['redacted_because']['sender'] : event.sender;
+                changes.push({changeType, event, sender, rule: previousState.unsigned.rule,
+                    ... previousState ? {previousState} : {} });
+                continue;
```

Comment:
I don't understand why we `continue` here.

Reply:
>Doesn't this deserve logging a warning?

Not really, it's normal for this to happen when we are wanting is for a rule to be removed.